packages/chain-bridge/src/adapters/LayerZeroAdapter.sol

```diff
@@ -170,6 +206,122 @@ contract LayerZeroAdapter is
                         EXTERNAL INTERFACE
     //////////////////////////////////////////////////////////////*/
 
+    /// @inheritdoc IAssetAdapter
+    /**
+     * @dev Transfers assets cross-chain using LayerZero OFT protocol.
+     *      Flow: pull tokens → approve OFT → send via OFT → refund excess native.
+     *      If message is provided, encodes as OFT compose for post-delivery execution.
+     * @param operationId Unique identifier for this transfer operation
+     * @param params Transfer parameters (asset, amount, destination, optional message)
+     */
+    function transferAsset(
+        bytes32 operationId,
+        BridgeTypes.ExecuteTransferParams calldata params,
+        BridgeTypes.BridgeOptions calldata
+    )
+        external
+        payable
+        onlyTrustedDestination(params.destinationChainId)
+        onlyRouter
+        nonReentrant
+    {
+        // 1. Validate and get OFT
+        address oft = _getOFTForAsset(params.asset, params.amount);
+
+        // 2. Pull tokens
+        IERC20(params.asset).safeTransferFrom(
+            msg.sender,
+            address(this),
+            params.amount
+        );
+
+        // 3. Get destination and cache EID
+        uint32 dstEid = _externalIdForChain(params.destinationChainId);
+        address dstAdapter = _getAdapterPeerOrRevert(params.destinationChainId);
+
+        // 4. Approve OFT
+        IERC20(params.asset).forceApprove(oft, params.amount);
+
+        // 5. Create compose message if needed
+        bytes memory composeMsg = params.message.length > 0
+            ? BridgeMessagingHelper.encodeRelayedTransferParams(
+                LayerZeroMessagingHelper.createRelayedTransferParams(
+                    params,
+                    operationId
+                )
+            )
+            : bytes("");
+
+        // 6. Create send parameters
+        SendParam memory sendParam = _createOFTSendParam(
+            dstEid,
+            dstAdapter,
+            params.amount,
+            composeMsg
+        );
+
+        // 7. Quote and send
+        MessagingFee memory fee = IOFT(oft).quoteSend(sendParam, false);
+        if (msg.value < fee.nativeFee) {
+            revert InsufficientFee(fee.nativeFee, msg.value);
+        }
+
+        IOFT(oft).send{value: fee.nativeFee}(
+            sendParam,
+            fee,
+            params.refundAddress
+        );
+
+        // 8. Refund excess
+        uint256 excess = msg.value > fee.nativeFee
+            ? msg.value - fee.nativeFee
+            : 0;
+        _refundExcessNative(params.refundAddress, excess);
```

Comment:
I think this function should be `_refundNative` as it is not calculating the excess and doesn't know anything about it. If you want to keep the name then do the excess calculation inside by passing the native fee instead